src/domain/sanity-content-transformer.ts

```
@@ -0,0 +1,201 @@
+ */
+export class SanityContentTransformer {
+  /**
+   * Transforms any WordPress content into a Sanity post content value object.
+   * This unified method handles both posts and pages when treating them as posts.
+   */
+  static async toSanityPost(wordpressContent: WordPressPost): Promise<SanityPostContent> {
+    // Convert HTML to BlockContent and extract media
+    const { content, media } = await htmlToBlockContent(wordpressContent.post_content)
+
+    // Create a plain text version of the content for the body field
+    const bodyText = this.extractPlainTextFromContent(content)
```

Biome (1.9.4)

[error] Using this in a static context can be confusing.

this refers to the class.
Unsafe fix: Use the class name instead.


(lint/complexity/noThisInStatic)